pex/platforms.py

```diff
@@ -62,33 +69,56 @@ def create(cls, platform):
 
     @staticmethod
     def _maybe_prefix_abi(impl, version, abi):
+        # type: (str, str, str) -> str
         if impl != "cp":
             return abi
         # N.B. This permits CPython users to pass in simpler extended platform
         # strings like `linux-x86_64-cp-27-mu` vs e.g. `linux-x86_64-cp-27-cp27mu`.
         impl_ver = "".join((impl, version))
         return abi if abi.startswith(impl_ver) else "".join((impl_ver, abi))
 
+    @classmethod
+    def from_tags(cls, platform, python, abi):
+        # type: (str, str, str) -> Platform
+        """Creates a platform corresponding to wheel compatibility tags.
+
+        See: https://www.python.org/dev/peps/pep-0425/#details
+        """
+        impl, version = python[:2], python[2:]
+        return cls(platform=platform, impl=impl, version=version, abi=abi)
+
     def __new__(cls, platform, impl, version, abi):
         if not all((platform, impl, version, abi)):
             raise cls.InvalidPlatformError(
-                "Platform specifiers cannot have blank fields. Given platform={platform!r}, impl={impl!r}, "
-                "version={version!r}, abi={abi!r}".format(
+                "Platform specifiers cannot have blank fields. Given platform={platform!r}, "
+                "impl={impl!r}, version={version!r}, abi={abi!r}".format(
                     platform=platform, impl=impl, version=version, abi=abi
                 )
             )
         platform = platform.replace("-", "_").replace(".", "_")
         abi = cls._maybe_prefix_abi(impl, version, abi)
         return super(Platform, cls).__new__(cls, platform, impl, version, abi)
 
-    @classmethod
-    def from_tags(cls, platform, python, abi):
-        """Creates a platform corresponding to wheel compatibility tags.
+    @property
+    def platform(self):
+        # type: () -> str
+        return super(Platform, self).platform  # type: ignore[no-any-return]
```

Comment:
You use cast below in `__str__` for str, you probably should for all of these too.